src/Shared/EnvironmentUtilities.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Build.Shared
+{
+    internal static partial class EnvironmentUtilities
+    {
+        public static bool Is64BitProcess => Marshal.SizeOf<IntPtr>() == 8;
+
+        //  Copied from .NET Framework source for Environment.Is64BitOperatingSystem (along with Win32Native APIs)
+        public static bool Is64BitOperatingSystem
+        {
+            get
+            {
+                bool isWow64; // WinXP SP2+ and Win2k3 SP1+
+                return Win32Native.DoesWin32MethodExist(Win32Native.KERNEL32, "IsWow64Process")
```

Comment:
@akoeplinger True, but I think we may as well leave it there to keep it in sync with the code for `Environment.Is64BitOperatingSystem`.